horovod/spark/lightning/remote.py

```diff
@@ -194,7 +195,10 @@ def on_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -
                       f"Val rows: {val_rows}, Val batch size: {val_batch_size}, Val_steps_per_epoch: {_val_steps_per_epoch}\n"
                       f"Checkpoint file: {remote_store.checkpoint_path}, Logs dir: {remote_store.logs_path}\n")
 
-            cuda_available = torch.cuda.is_available()
+            if not should_pin_gpu and verbose:
+                print("Skip pinning current process to the GPU.")
+
+            cuda_available = torch.cuda.is_available() and should_pin_gpu
```

Comment:
It should be mentioned in the API that `pin_gpu==False` disables CUDA.

Here should be a **warning** that though CUDA is available is is not used as pin_gpu is false. Would otherwise be hard to find out why CUDA is not used.

Reply:
makes sense, added a check to emit message when cuda is available but pin_gpu=false